packages/notebook-extension/src/index.ts

```diff
@@ -1911,9 +1911,39 @@ function addCommands(
     }
   );
 
+  tracker.selectionChanged.connect(() => {
+    commands.notifyCommandChanged(CommandIDs.duplicateBelow);
+    commands.notifyCommandChanged(CommandIDs.deleteCell);
+    commands.notifyCommandChanged(CommandIDs.copy);
+    commands.notifyCommandChanged(CommandIDs.cut);
+    commands.notifyCommandChanged(CommandIDs.pasteBelow);
+    commands.notifyCommandChanged(CommandIDs.pasteAbove);
+    commands.notifyCommandChanged(CommandIDs.pasteAndReplace);
+    commands.notifyCommandChanged(CommandIDs.moveUp);
+    commands.notifyCommandChanged(CommandIDs.moveDown);
+    commands.notifyCommandChanged(CommandIDs.run);
+    commands.notifyCommandChanged(CommandIDs.runAll);
+    commands.notifyCommandChanged(CommandIDs.runAndAdvance);
+    commands.notifyCommandChanged(CommandIDs.runAndInsert);
+  });
+
   commands.addCommand(CommandIDs.runAndAdvance, {
-    label: trans.__('Run Selected Cells'),
-    caption: trans.__('Run the selected cells'),
+    label: args => {
+      const current = getCurrent(tracker, shell, args);
+      return trans._n(
+        'Run Selected Cell',
+        'Run Selected Cells',
+        current?.content.select.length ?? 0
+      );
+    },
+    caption: args => {
+      const current = getCurrent(tracker, shell, args);
+      return trans._n(
+        'Run this cell and advance',
+        `Run these ${current?.content.selectedCells.length} cells and advance`,
```

Comment:
The special placeholder `%1` will be replaced by the number provided to determine if singular or plural is to be displayed.

```suggestion
        'Run these %1 cells and advance',
```